gradio/components.py

```diff
@@ -247,6 +247,7 @@ def __init__(
         default_value: str = "",
         *,
         lines: int = 1,
+        max_lines: int = 100,
```

Comment:
How about we leave `max_lines` as `None` by default, which allows the Textbox to get arbitrarily large. And set its type to be `Optional[int]`

Reply:
I had the same intention, my arbitrarily large number was 100, what is your arbitrarily large number? Frontend will have its own max value, so we can give a default value of arbitrarily large number.

Reply:
Why does there need to be a max value set in the frontend? I would set the default for `max_lines` to be `None`, meaning that there is no max -- the textbox just keeps getting longer.

Reply:
@pngwn I think we would not let Textbox grow infinitely large, that's what I understood from our discussions and how I feel. 
What are your thoughts on this @pngwn?

Reply:
I think allowing the textbox to grow infinitely large is a poor default, we can easily set a default in the frontend though without there being one in the back.

Reply:
What about allowing 1 million max_lines if the user prefers. I think we would have a max for that as well right?

Reply:
Btw this design assigns a max value(100) when the user does not specify it, we could increase it if we wish. I don't see a need for passing None to frontend and frontend selecting that max value.

Reply:
@FarukOzderim We dont' need to pass anything to the front end particularly, the absence of any preference can trigger a default. I don't think UX defaults should live in the python library particularly.

20 is a better default. We don't want to a hard maximum because people may want longer 'prose'.